File: src/main/java/org/cimsbioko/search/SearchQueue.java
package org.cimsbioko.search;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class SearchQueue {

    private ExecutorService execService = Executors.newSingleThreadExecutor();
    private boolean shutdown;

    public void shutdown() {
        this.shutdown = true;
        execService.shutdownNow();
    }

    public boolean isShutdown() {
        return shutdown;
    }

    public void queue(SearchJob job) {
        if (!shutdown) {
            job.setQueue(this);
            execService.submit(job);
        }
    }
}

File: src/main/java/org/cimsbioko/search/SearchQueue.kt
package org.cimsbioko.search

import java.util.concurrent.Executors

class SearchQueue {

    private val execService = Executors.newSingleThreadExecutor()

    var isShutdown = false
        private set

    fun shutdown() {
        execService.shutdownNow().also { isShutdown = true }
    }

    fun queue(job: SearchJob) {
        if (!isShutdown) {
            execService.submit(job.also { it.setQueue(this) })
        }
    }
}